### ISSUE TITLE
QTL error: Promise already under evaluation: recursive default argument reference or earlier problems?

### ISSUE THREAD

**Issue by thisisamirv** [fischuu/GenomicTools#10]
When I run the following code from the book:

`phenoQTL <- QTL(
    pheno = t(pheno_data[2, ]),
    geno = geno_data,
  )`

I get this error: Promise already under evaluation: recursive default argument reference or earlier problems?

**Comment by fischuu** [comment]
Could you create a minimal example that reproduces the error? How do your objects `pheno_data` and `geno_data` look like?

**Comment by fischuu** [comment]
I could reproduce the error also with my testing dataset - I will look into it and upload a bugfix as soon as I traced the reason.

**Comment by thisisamirv** [comment]
I found the reason already and fixed it in my pull request. It was due to setting variables in functions equal to themselves. For example:

function(m = m, n = n, ....

**Comment by fischuu** [comment]
Yes, actually, I noticed that as well. And then, while looking for it, I noticed another problem that might occur, when the user has imported the vcf such that the identifier for missing data is not identified properly then the calculation of the MAF might go wrong.

I just noticed the pull request as well, thanks for that! I think your editor changed quite many places for better readability of the code, so it might take a little while to go through the changes and trace your edits, sorry!
